### PR TITLE
lifted com.typesafe's config library to 1.3.0 and Java8

### PR DESCRIPTION
**Pull request by mreif**
The wrapped config library has been updated to a new (compatible) version. Tests still run.

### CODE REVIEW

**Comment by mreif** [comment]
Don't know how to fix this, does work for me in every setting.

**Comment by kailuowang** [comment]
Thanks for the PR, it seems that typesafe config upgraded to java 8, if you update the java version here, including Travis setting, it might pass.

**Comment by mreif** [comment]
I don't know travis very well do you know what I have to specify?

**Comment by kailuowang** [comment]
in .travis.yml you'll find jdk settings for running on travis
also it's probably a good idea to enforce java8 in the build.sbt by
adding `-target:jvm-1.8` to the existing scalaOptions

**Comment by kailuowang** [comment]
Looks like that only the new trusty env supports openjdk8 
https://docs.travis-ci.com/user/trusty-ci-environment/#JVM
we can try use that by adding the following to .travis.yml
```
sudo: required
dist: trusty
```

**Comment by mreif** [comment]
The build fails when the jvm-switcher tries to switch to the openjdk8: accoring to the documention of this repository: [jdk_switcher]()https://github.com/michaelklishin/jdk_switcher) openjdk8 is not supported. Even if the above commit messagse said that this support has been added. We might have to adapt the script for that.

File: build.sbt [hunk 1]
```diff
@@ -30,9 +30,11 @@ scalacOptions <++= scalaVersion map { sv =>
     "-feature",
     "-language:postfixOps",
     "-language:implicitConversions",
-    "-language:higherKinds"
+    "-language:higherKinds",
+    "-target:jvm-1.8"
```
Comment (kailuowang) [inline build.sbt]: Let's leave changes to scala 2.10 to the other 1.0.x  branch (and we don't have to update that one). Basically this line is unnecessary.

**Comment by kailuowang** [comment]
Thank you @mreif.

**Pull request merged**